cli/templateedit.go

```diff
@@ -98,71 +56,51 @@ func (r *RootCmd) templateEdit() *clibase.Cmd {
+				autostopRequirementWeeks:       autostopRequirementWeeks,
+				autostartRequirementDaysOfWeek: autostartRequirementDaysOfWeek,
+				failureTTL:                     failureTTL,
+				dormancyThreshold:              dormancyThreshold,
+				dormancyAutoDeletion:           dormancyAutoDeletion,
+				allowUserCancelWorkspaceJobs:   allowUserCancelWorkspaceJobs,
+				allowUserAutostart:             allowUserAutostart,
+				allowUserAutostop:              allowUserAutostop,
+				requireActiveVersion:           requireActiveVersion,
+			})
+			if err != nil {
+				return err
 			}
 
-			var deprecated *string
-			if !userSetOption(inv, "deprecated") {
-				deprecated = &deprecationMessage
-			}
+			req := updateTemplateMetaRequest(updateTemplateMetaArgs{
+				client:                             client,
+				inv:                                inv,
+				template:                           template,
+				unsetAutostopRequirementDaysOfWeek: unsetAutostopRequirementDaysOfWeek,
 
-			req := codersdk.UpdateTemplateMeta{
-				Name:             name,
-				DisplayName:      displayName,
-				Description:      description,
-				Icon:             icon,
-				DefaultTTLMillis: defaultTTL.Milliseconds(),
-				MaxTTLMillis:     maxTTL.Milliseconds(),
-				AutostopRequirement: &codersdk.TemplateAutostopRequirement{
-					DaysOfWeek: autostopRequirementDaysOfWeek,
-					Weeks:      autostopRequirementWeeks,
-				},
-				AutostartRequirement: &codersdk.TemplateAutostartRequirement{
-					DaysOfWeek: autostartRequirementDaysOfWeek,
-				},
-				FailureTTLMillis:               failureTTL.Milliseconds(),
-				TimeTilDormantMillis:           dormancyThreshold.Milliseconds(),
-				TimeTilDormantAutoDeleteMillis: dormancyAutoDeletion.Milliseconds(),
-				AllowUserCancelWorkspaceJobs:   allowUserCancelWorkspaceJobs,
-				AllowUserAutostart:             allowUserAutostart,
-				AllowUserAutostop:              allowUserAutostop,
-				RequireActiveVersion:           requireActiveVersion,
-				DeprecationMessage:             deprecated,
-			}
+				name:                           name,
+				displayName:                    displayName,
+				description:                    description,
+				icon:                           icon,
+				defaultTTL:                     defaultTTL,
+				maxTTL:                         maxTTL,
+				autostopRequirementDaysOfWeek:  autostopRequirementDaysOfWeek,
+				autostopRequirementWeeks:       autostopRequirementWeeks,
+				autostartRequirementDaysOfWeek: autostartRequirementDaysOfWeek,
+				failureTTL:                     failureTTL,
+				dormancyThreshold:              dormancyThreshold,
+				dormancyAutoDeletion:           dormancyAutoDeletion,
+				allowUserCancelWorkspaceJobs:   allowUserCancelWorkspaceJobs,
+				allowUserAutostart:             allowUserAutostart,
+				allowUserAutostop:              allowUserAutostop,
+				requireActiveVersion:           requireActiveVersion,
+				deprecationMessage:             deprecationMessage,
+			})
```

Comment:
These mega structs are big code smells. It's not your PR's fault, but would enjoy seeing someone clean it up.

Reply:
On second thought, why don't we use the `codersdk` types when validating entitlements?